### ISSUE TITLE
Add label to every metric in cluster, while using serviceMonitors. 

### ISSUE THREAD

**Issue by BenCoughlan15** [prometheus-operator/prometheus-operator#6139]
Proposal
Hey, I'm currently using the prometheus operator, and kube-prometheus-stack chart.

I currently have about 120 serviceMonitors in each cluster, and 12 clusters.

I then port all the metrics from those clusters to a central global mimir cluster.

I currently use the externalLabel parameter to add the cluster-id label to all the metrics, and this works fine.

However, in Grafana I now have "global" dashboards, that are aware of the cluster-id label, and regional ones that don't carry the cluster-id label.

I'd rather just have one dashboard for both Global, and Regional, as it's a lot less to maintain.

So my choices are:

1: Update every serviceMonitor, across the whole estate, which means bumping a lot of helm charts, which is a lot of work.
2: Maintain two types of dashboard.

I guess, my question is:
Is there any way to have a global setting to add a label to all metrics? I've searched everywhere, and can't seem to find it.

If not, then would it make sense to have that added as a feature?

**Comment by simonpasquier** [comment]
I guess that you're looking for something like https://github.com/prometheus-operator/prometheus-operator/issues/5947?

**Comment by simonpasquier** [comment]
Agreed with https://github.com/prometheus-operator/prometheus-operator/issues/6139#issuecomment-2039255165. @BenCoughlan15 feel free to re-open if it's not the case.
